plugins/experimental/uri_signing/parse.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "uri_signing.h"
+#include "parse.h"
+#include "config.h"
+#include "jwt.h"
+#include "cookie.h"
+#include "timing.h"
+#include <cjose/cjose.h>
+#include <jansson.h>
+#include <string.h>
+#include <ts/ts.h>
+#include <inttypes.h>
+
+cjose_jws_t *
+get_jws_from_query(const char *uri, size_t uri_ct, const char *paramName)
+{
+  PluginDebug("Parsing JWS from query string: %.*s", (int)uri_ct, uri);
+  const char *query = uri;
+  const char *end   = uri + uri_ct;
+  while (query != end && *query != '?')
+    ++query;
```

Comment:
Why pre-increment?

Reply:
Why not?

If it doesn't matter, I usually use pre-increment, just because it has no strange unexpected behaviour. It's a tiny reduction on cognitive load. Here, either works trivially.